docs: add documentation for user group syncing (#2314)
Unleash/unleash: website/docs/user_guide/rbac.md

- a **name** (required)
- a **description** (optional)
- a list of users (optional)
- a list of SSO groups to sync from (optional)

Groups do nothing on their own. They must be given a role on a project to assign permissions. You can assign both standard roles and custom project roles to groups.

While a user can only have one role in a given project, a user may belong to multiple groups, and each of those groups may be given a role on a project. In the case where a given user is given permissions to a project through more than one group, the user will inherit most permissive permissions of all their groups in that project.

## User Group SSO Integration

:::info availability

User group syncing is planned to be released in Unleash 4.18 and will be available for enterprise customers.

:::

> User groups also support integration with your Single Sign-On (SSO) provider. This allows you to automatically assign users to groups when they log in through SSO. Check out [_how to set up group SSO sync_](../how-to/how-to-set-up-group-sso-sync.md) for more details.

Reviewer: Wonder if we can clarify this a bit? So that it's not generic "details" (because they should be in the reference)?

```suggestion
User groups also support integration with your Single Sign-On (SSO) provider. This allows you to automatically assign users to groups when they log in through SSO. Check out [_how to set up group SSO sync_](../how-to/how-to-set-up-group-sso-sync.md) for a step-by-step walkthrough.
```

Author: Sure, that looks better to me

Users that have been added to a group through your SSO provider will be automatically removed next time they log in if they've been removed from the SSO group. Users that have been manually added to the group will not be affected.

To enable group sync, you'll need to set two fields in your SSO provider configuration options:

- **enable group syncing**:

>   Turns on group syncing, by default this is disabled.

Reviewer: ```suggestion
  Turns on group syncing. This is disabled by default.
```

- **group field JSON path**

  A JSON path that should point to the groups field in your token response. This should match the exact field returned by the provider. For example, if your token looks like this:

  ```json
  {

    "iss": "https://some-url.com",
    "azp": "1234987819200.apps.some-url.com",
    "aud": "1234987819200.apps.some-url.com",
    "sub": "10769150350006150715113082367",
    "at_hash": "HK6E_P6Dh8Y93mRNtsDB1Q",
    "hd": "example.com",
    "email": "jsmith@example.com",
    "email_verified": "true",
    "groups": ["test-group", "test-group-2"], //the field where groups are specified
    "iat": 1353601026,
    "exp": 1353604926,
    "nonce": "0394852-3190485-2490358"
  }
  ```

>   You need to set the `Group Field JSON path` to "groups".

Reviewer: I'm not sure about the use of quotes here, but I feel like regular double quotes may be the way to go.

```suggestion
  You need to set the "Group Field JSON path" to "groups".
```

Groups do nothing on their own. They must be given a role on a project to assign permissions. You can assign both standard roles and custom project roles to groups.

While a user can only have one role in a given project, a user may belong to multiple groups, and each of those groups may be given a role on a project. In the case where a given user is given permissions to a project through more than one group, the user will inherit most permissive permissions of all their groups in that project.

> ## User Group SSO Integration

Reviewer: This is a nice paragraph, but I'm lacking information on what the options are for syncing. The how-to guides reference the "Group Field JSON path" thing. What does that mean? Are there other things that need to be configured? This is the place to introduce and explain them!  😄

Author: I've added a bunch of text, will you check it out?

Reviewer: Sure will!

:::info availability

> User group SSO Integration is planned to be released in **Unleash 4.18** and will be available for enterprise customers.

Reviewer: Same thing again regarding availability; this is just a placeholder for attention.

Author: Patched this

:::

> User groups also support integration with your Single Sign On provider. This allows you to automatically assign users to groups when they login through Single Sign On (SSO). See [_how to set up group sso sync_](../how-to/how-to-set-up-group-sso-sync.md) for more details.

Reviewer: Isn't it "single sign-on" with a hyphen?

```suggestion
User groups also support integration with your Single Sign-On provider. This allows you to automatically assign users to groups when they log in through Single Sign-On (SSO). Check out [_how to set up group SSO sync_](../how-to/how-to-set-up-group-sso-sync.md) for more details.
```

Author: Good spot, I've added your suggestion but also moved (SSO) to the first instance here
